pkg/protocol/uri.go

```diff
@@ -480,6 +480,15 @@ func splitHostURI(host, uri []byte) ([]byte, []byte, []byte) {
 }
 
 func normalizePath(dst, src []byte) []byte {
+	// replace all backslashes with forward slashes
+	for {
+		n := bytes.Index(src, bytestr.StrBackSlash)
```

Comment:
Seems it will double check for bytes that have been replaced, is this an performance issue on long paths?

Reply:
> 1. Seems it will double check for bytes that have been replaced, is this an performance issue on long paths?
> 2. I'm not sure it's a good idea to replace `src` in place, maybe it's hard to understand that if `normalizePath` has a return value and also modifies the input parameters.

maybe this one would be better

```go
func normalizePath(dst, src []byte) []byte {
	dst = dst[:0]
	dst = addLeadingSlash(dst, src)
	dst = decodeArgAppendNoPlus(dst, src)

	// replace all backslashes with forward slashes
	for i := range dst {
		if dst[i] == '\\' {
			dst[i] = '/'
		}
	}
	...
}
```

Reply:
`bytes.IndexByte` will do a SIMD accelerate in some platforms. Not sure which one is better.

And since the bug only happens in Windows? Maybe add a `if filepath.Separator == '\\' {` to filter the platform?